lib/src/MGPOptional.ts

```diff
@@ -43,59 +43,103 @@ export class MGPOptional<T> {
         };
     }
 
-    private constructor(private readonly value: T | null) {}
+    public abstract isPresent(): boolean;
```

Comment:
"boolean" ->  "this is MGPOptionalPresent<T>" ? "same" for isAbsent

Reply:
On est d'accord que c'est louche car
TypeGenerique a = monInstanceSpecifique doit marcher 
et que
TypeSpecifique a = monInstanceGeneriqueOuCousineDeTypeSpecifique doit PAS marcher
Mais que là même le premier ne marche pas ?

Reply:
On est d'accord oui !!